### PR TITLE
feat(quantic)!: Change names of DOM Events used, to add the 'quantic__' prefix

### PR DESCRIPTION
**Pull request by SimonMilord**
[SFINT-5351](https://coveord.atlassian.net/browse/SFINT-5351)

### IN THIS PR: 
Changed the prefix of custom DOM events in the quantic package to have the ‘quantic__’  prefix so that the naming convention is standard across the package.

### TESTS:
image

[SFINT-5351]: https://coveord.atlassian.net/browse/SFINT-5351?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticFeedback/quanticFeedback.js [hunk 1]
```diff
@@ -94,38 +94,38 @@ export default class QuanticFeedback extends LightningElement {
...
 
   /**
-   * Fires the "like" event.
+   * Fires the "quantic__like" event.
    * @returns {void}
    */
   handleLike() {
     this.dispatchEvent(
-      new CustomEvent(`like`, {
+      new CustomEvent(`quantic__like`, {
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticFeedback/quanticFeedback.js:102]: These events haven't been updated where they were listened to:

Example
https://github.com/coveo/ui-kit/blob/a6dd19c883664e09a86dd80c81c55e637315ac8e/packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/templates/generatedAnswer.html#L14
https://github.com/coveo/ui-kit/blob/a6dd19c883664e09a86dd80c81c55e637315ac8e/packages/quantic/force-app/main/default/lwc/quanticSmartSnippet/quanticSmartSnippet.html#L31

Search for "onlike", "ondislike", "onpressexplainwhy"

File: packages/quantic/force-app/main/default/lwc/quanticFoldedResultList/quanticFoldedResultList.js [hunk 2]
```diff
@@ -134,7 +134,7 @@ export default class QuanticFoldedResultList extends LightningElement {
 
   registerTemplates() {
     this.dispatchEvent(
-      new CustomEvent('registerresulttemplates', {
+      new CustomEvent('quantic__registerresulttemplates', {
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticFoldedResultList/quanticFoldedResultList.js:137]: search for `onregisterresulttemplates` to find places where this need to be updated

File: packages/quantic/force-app/main/default/lwc/quanticNumberButton/quanticNumberButton.js [hunk 3]
```diff
@@ -40,6 +40,8 @@ export default class QuanticNumberButton extends LightningElement {
   }
 
   select() {
-    this.dispatchEvent(new CustomEvent('select', {detail: this.number}));
+    this.dispatchEvent(
+      new CustomEvent('quantic__select', {detail: this.number})
+    );
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticNumberButton/quanticNumberButton.js:45]: https://github.com/coveo/ui-kit/blob/a6dd19c883664e09a86dd80c81c55e637315ac8e/packages/quantic/force-app/main/default/lwc/quanticPager/quanticPager.html#L20-L21

File: packages/quantic/force-app/main/default/lwc/quanticPill/quanticPill.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ export default class QuanticPill extends LightningElement {
   @api actionName;
 
   deselect() {
-    this.dispatchEvent(new CustomEvent('deselect'));
+    this.dispatchEvent(new CustomEvent('quantic__deselect'));
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticPill/quanticPill.js:37]: Search for where the `<c-quantic-pill>` is used to see where possibly we have the `ondeselect={}`

File: packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js [hunk 5]
```diff
@@ -145,7 +145,7 @@ export default class QuanticRadioButtonsGroup extends LightningElement {
     this._value = value;
 
     this.dispatchEvent(
-      new CustomEvent('change', {
+      new CustomEvent('quantic__change', {
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/quanticRadioButtonsGroup.js:148]: `<c-quantic-radio-buttons-group onchange={}>` search for that

File: packages/quantic/force-app/main/default/lwc/quanticResultQuickview/quanticResultQuickview.js [hunk 6]
```diff
@@ -196,7 +199,7 @@ export default class QuanticResultQuickview extends LightningElement {
 
   dispatchHasPreview(hasPreview) {
     this.dispatchEvent(
-      new CustomEvent('haspreview', {
+      new CustomEvent('quantic__haspreview', {
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticResultQuickview/quanticResultQuickview.js:202]: This was not updated in the component that listens to this event:

https://github.com/coveo/ui-kit/blob/a6dd19c883664e09a86dd80c81c55e637315ac8e/packages/quantic/force-app/main/default/lwc/quanticResult/quanticResult.js#L68

File: packages/quantic/force-app/main/default/lwc/quanticStatefulButton/quanticStatefulButton.js [hunk 7]
```diff
@@ -68,9 +68,9 @@ export default class QuanticStatefulButton extends LightningElement {
   handleClick(event) {
     event.stopPropagation();
     if (this.selected) {
-      this.dispatchEvent(new CustomEvent('deselect'));
+      this.dispatchEvent(new CustomEvent('quantic__deselect'));
     } else {
-      this.dispatchEvent(new CustomEvent('select'));
+      this.dispatchEvent(new CustomEvent('quantic__select'));
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticStatefulButton/quanticStatefulButton.js:73]: Search for `<c-quantic-stateful-button onselect={} ondeselect={}>`

File: packages/quantic/force-app/main/default/lwc/quanticTabBar/quanticTabBar.js [hunk 8]
```diff
@@ -39,7 +39,7 @@ export default class QuanticTabBar extends LightningElement {
   connectedCallback() {
     window.addEventListener('click', this.closeDropdown);
     window.addEventListener('resize', this.updateTabsDisplay);
-    this.addEventListener('tab_rendered', this.updateTabsDisplay);
+    this.addEventListener('quantic__tab_rendered', this.updateTabsDisplay);
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticTabBar/quanticTabBar.js:42]:
```suggestion
    this.addEventListener('quantic__tabrendered', this.updateTabsDisplay);
```

File: packages/quantic/force-app/main/default/lwc/quanticTab/quanticTab.js [hunk 9]
```diff
@@ -68,7 +68,9 @@ export default class QuanticTab extends LightningElement {
 
   renderedCallback() {
     initializeWithHeadless(this, this.engineId, this.initialize);
-    this.dispatchEvent(new CustomEvent('tab_rendered', {bubbles: true}));
+    this.dispatchEvent(
+      new CustomEvent('quantic__tab_rendered', {bubbles: true})
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticTab/quanticTab.js:72]:
```suggestion
      new CustomEvent('quantic__tabrendered', {bubbles: true})
```

**Review by erocheleau — CHANGES_REQUESTED** [review]
Many missing refactoring of the events names

File: packages/quantic/force-app/examples/main/lwc/actionSendRating/actionSendRating.js [hunk 10]
```diff
@@ -9,7 +9,7 @@ export default class ActionSendRating extends LightningElement {
     }, 0);
   }
   handleSendRating() {
-    const customEvent = new CustomEvent('rating', {
+    const customEvent = new CustomEvent('quantic__rating', {
```
Comment (erocheleau) [inline packages/quantic/force-app/examples/main/lwc/actionSendRating/actionSendRating.js:12]: I think the event names in the examples/ I would leave alone

**Review by erocheleau — CHANGES_REQUESTED** [review]
You're also missing this one :

https://github.com/coveo/ui-kit/blob/e338511e7338e9fdc87a46a392835207baf82a0d/packages/quantic/force-app/main/default/lwc/quanticSearchBoxSuggestionsList/quanticSearchBoxSuggestionsList.js#L256

File: packages/quantic/force-app/examples/main/lwc/actionAddFacets/actionAddFacets.js [hunk 11]
```diff
@@ -10,8 +10,8 @@ export default class ActionAddFacets extends LightningElement {
 
   handleAddFacets() {
     const eventName = this.withoutInputs
-      ? 'addFacetsWithoutInputs'
-      : 'addFacets';
+      ? 'quantic__addFacetsWithoutInputs'
+      : 'quantic__addFacets';
```
[the remark this answers is not in the document]
Comment (SimonMilord) [reply packages/quantic/force-app/examples/main/lwc/actionAddFacets/actionAddFacets.js:14]: they seem to be used for E2E tests. The add Facet was added to check whether the refine toggle is disabled if there are no tab/filters to show in the filters tab

File: packages/quantic/force-app/examples/main/lwc/actionSendRating/actionSendRating.js (hunk 10 above)
Comment (SimonMilord) [reply packages/quantic/force-app/examples/main/lwc/actionSendRating/actionSendRating.js:12]: will revert for examples

File: packages/quantic/force-app/examples/main/lwc/exampleQuanticRefineToggle/exampleQuanticRefineToggle.js [hunk 12]
```diff
@@ -13,17 +13,17 @@ export default class ExampleQuanticRefineToggle extends LightningElement {
   facetWithoutInputs = false;
 
   connectedCallback() {
-    this.addEventListener('addFacets', this.handleAddFacets);
+    this.addEventListener('quantic__addFacets', this.handleAddFacets);
```
[the remark this answers is not in the document]
Comment (SimonMilord) [reply packages/quantic/force-app/examples/main/lwc/exampleQuanticRefineToggle/exampleQuanticRefineToggle.js:16]: will revert, but I answered in another comment

File: packages/quantic/force-app/main/default/lwc/quanticDocumentSuggestion/quanticDocumentSuggestion.js [hunk 13]
```diff
@@ -84,7 +84,7 @@ export default class QuanticDocumentSuggestion extends LightningElement {
     this.validateProps();
     if (!this.hasInitializationError) {
       registerComponentForInit(this, this.engineId);
-      this.template.addEventListener('rating', this.onRating);
+      this.template.addEventListener('quantic__rating', this.onRating);
```
[the remark this answers is not in the document]
Comment (SimonMilord) [reply packages/quantic/force-app/main/default/lwc/quanticDocumentSuggestion/quanticDocumentSuggestion.js:87]: in `quanticDocumentSuggestion.js`

File: packages/quantic/force-app/examples/main/lwc/actionSendRating/actionSendRating.js (hunk 10 above)
Comment (mmitiche) [reply packages/quantic/force-app/examples/main/lwc/actionSendRating/actionSendRating.js:12]: Did we decide to revert the events in the */examples as they are not part of the package? @SimonMilord

File: packages/quantic/force-app/main/default/lwc/quanticCardContainer/quanticCardContainer.js [hunk 14]
```diff
@@ -29,14 +29,14 @@ export default class QuanticCardContainer extends LightningElement {
...
-    const headerClickEvent = new CustomEvent('headerclick', {});
+    const headerClickEvent = new CustomEvent('quantic__headerclick', {});
     this.dispatchEvent(headerClickEvent);
   }
 
   handleHeaderKeyDown(evt) {
     if (evt.code === 'Enter' || evt.code === 'Space') {
       evt.preventDefault();
-      const headerKeyDownEvent = new CustomEvent('headerkeydown', {});
+      const headerKeyDownEvent = new CustomEvent('quantic__headerkeydown', {});
```
[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticCardContainer/quanticCardContainer.js:39]: This event is not present in the js doc please add it.

File: packages/quantic/force-app/main/default/lwc/quanticDocumentSuggestion/quanticDocumentSuggestion.js (hunk 13 above)
[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticDocumentSuggestion/quanticDocumentSuggestion.js:87]: This event is used/dispatched by the `rating` slot that can be passed to the QuanticdocumentSuggestion component.

File: packages/quantic/force-app/main/default/lwc/quanticFeedback/quanticFeedback.js [hunk 15]
```diff
@@ -6,9 +6,9 @@ import {LightningElement, api} from 'lwc';
 
 /**
  * The `QuanticFeedback` component displays a question and two buttons for giving positive or negative feedback.
- * @fires CustomEvent#like
- * @fires CustomEvent#dislike
- * @fires CustomEvent#pressExplainWhy
+ * @fires CustomEvent#quantic__like
+ * @fires CustomEvent#quantic__dislike
+ * @fires CustomEvent#quantic__pressExplainWhy
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticFeedback/quanticFeedback.js:11]: The name convention that currently be used is to have all letters in lower case.

Comment (SimonMilord) [reply packages/quantic/force-app/main/default/lwc/quanticFeedback/quanticFeedback.js:11]: will update that

**Comment by mmitiche** [comment]
@SimonMilord  Please address the conflicts and confirm that the E2E are all passing correctly ✅
Ping me directly right after so I approve,

**Review by mmitiche — APPROVED** [review]
👍

**Review by erocheleau — APPROVED** [review]
I trust your second pass 😁

**Pull request merged**